systest/multi-tenancy/basic_test.go

```diff
@@ -76,53 +86,249 @@ func TestAclBasic(t *testing.T) {
 			}
 		}
 	`
-	resp := testutil.QueryData(t, dc, query)
+	gcli, suite.cleanup, err = suite.dc.Client()
+	require.NoError(t, err)
+	require.NoError(t, gcli.LoginIntoNamespace(context.Background(), "groot", "password", ns))
+	resp := suite.QueryData(gcli, query)
 	testutil.CompareJSON(t,
 		`{"me": [{"name":"guy1","nickname":"RG"},
 		{"name": "guy2", "nickname":"RG2"}]}`,
 		string(resp))
 
 	// groot of namespace 0 should not see the data of namespace-1
-	dc = testutil.DgClientWithLogin(t, "groot", "password", 0)
-	resp = testutil.QueryData(t, dc, query)
+	gcli, suite.cleanup, err = suite.dc.Client()
+	require.NoError(t, err)
+	require.NoError(t, gcli.LoginIntoNamespace(context.Background(), "groot", "password", 0))
+	resp = suite.QueryData(gcli, query)
 	testutil.CompareJSON(t, `{"me": []}`, string(resp))
 
 	// Login to namespace 1 via groot and create new user alice.
-	token, err := testutil.Login(t, &testutil.LoginParams{UserID: "groot", Passwd: "password", Namespace: ns})
-	require.NoError(t, err, "login failed")
-	testutil.CreateUser(t, token, "alice", "newpassword")
+	hcli, err = suite.dc.HTTPClient()
+	require.NoError(t, err)
+	err = hcli.LoginIntoNamespace("groot", "password", ns)
+	require.NotNil(t, hcli, "token for the namespace is nil")
+	require.NoErrorf(t, err, "login with namespace %d failed", ns)
+	_, err = hcli.CreateUser("alice", "newpassword")
+	require.NoError(t, err)
 
 	// Alice should not be able to see data added by groot in namespace 1
-	dc = testutil.DgClientWithLogin(t, "alice", "newpassword", ns)
-	resp = testutil.QueryData(t, dc, query)
+	gcli, suite.cleanup, err = suite.dc.Client()
+	require.NoError(t, err)
+	require.NoError(t, gcli.LoginIntoNamespace(context.Background(), "alice", "newpassword", ns))
+	resp = suite.QueryData(gcli, query)
 	testutil.CompareJSON(t, `{}`, string(resp))
 
 	// Create a new group, add alice to that group and give read access of <name> to dev group.
-	testutil.CreateGroup(t, token, "dev")
-	testutil.AddToGroup(t, token, "alice", "dev")
-	testutil.AddRulesToGroup(t, token, "dev",
-		[]testutil.Rule{{Predicate: "name", Permission: acl.Read.Code}}, true)
+	err = hcli.CreateGroup("dev")
```

Comment:
you could just do this in one line, something like
require.NoError(t, hcli.CreateGroup("dev"))